calico-vpp-agent/cni/cni_server.go

```diff
@@ -315,9 +324,57 @@ func (s *Server) rescanState() {
 		default:
 			s.log.Errorf("Interface add failed %s : %v", podSpecCopy.String(), err)
 		}
+		if s.PodNeedsRedirectToHost(&podSpecCopy) {
+			err := s.AddRedirectToHostToInterface(podSpecCopy.TunTapSwIfIndex, 30)
+			if err != nil {
+				s.log.Error(err)
+			}
+		}
+	}
+}
+
+func (s *Server) PodNeedsRedirectToHost(podSpec *storage.LocalPodSpec) bool {
+	return strings.HasPrefix(podSpec.WorkloadID, "kube-system/coredns")
+}
```

Comment:
No need to restrict this just to the coredns, arguably all rules should apply to all pods